rules/__tests__/prefer_to_be_defined.test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester;
+const rules = require('../../').rules;
+
+const ruleTester = new RuleTester();
+
+ruleTester.run('prefer_to_be_defined', rules['prefer-to-be-defined'], {
+  valid: ['expect(true).toBeDefined();'],
+
+  invalid: [
+    {
+      code: 'expect(true).not.toBe(undefined);',
```

Comment:
See https://github.com/jest-community/eslint-plugin-jest/pull/8#discussion_r150161191, I don't think this rule should warn on `toBeUndefined()`, the other rule can warn on that